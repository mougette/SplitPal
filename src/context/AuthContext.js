import React, { useState} from "react";
import createDataContext from './createDataContext';
import {Post} from '../components/RestPost';
import {Get} from '../components/RestGet';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'signout':
      return {token: null, email: ''};
    case 'signin':
    return {
            email: action.payload.email,
            token: action.payload.token,
          };
    case 'signup':
      return {
        token: action.payload.token,
        email: action.payload.email,
      };
  }
};

const signup = dispatch => {
    const [account, setAccount] = useState("");
  return ({email, password, password2, firstName, lastName, phoneNumber}) => {
    Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login",
                JSON.stringify({
                 email: email,
                 password: password,
                 password2: password2,
                    firstName: firstName,
                    lastName: lastName,
                    phoneNumber: phoneNumber,
                 }))
        .then(response => {
    console.log("Account: " + account + " Account End");
      if(response == "Account Successfully Made") {
              dispatch({
                  type: 'signup',
                  payload: {
                      token: 'some access token here',
                      email: email,
                  },
              });
      }
      else {
              alert(response)
      }
      }
      );


  };
};

const signin = dispatch => {
  return ({email, password}) => {
    // Do some API Request here
   Get("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/login","?email="+email+"&password="+password)
   .then(response =>{
    if(response == "User info correct"){
    dispatch({
      type: 'signin',
      payload: {
        token: 'some access token here',
        email: email,
      },
    });
    }});
  };
};

const signout = dispatch => {
  return () => {
    dispatch({type: 'signout'});
  };
};

export const {Provider, Context} = createDataContext(
  authReducer,
  {signin, signout, signup},
  {token: null, email: ''},
);
