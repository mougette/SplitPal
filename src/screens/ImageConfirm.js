import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { StyleSheet, Text, View, Image, Button, Navigation } from 'react-native';

const ImageConfirm = ({navigation},props) => {
  return (
    console.log(props.image)
    <View style = {styles.master}>
        <View style={{height: 300, flex: 1,}}>
            <View style={styles.container}>
                <Image
                    resizeMode={'contain'}
                    style={styles.fill}
                    source={props.image}
                    />

            </View>
            <View style={{height: 40,}}>
                <Button title = "Confirm" onPress = { () => navigation.navigate("SplitGroup") }/>
            </View>
            <View style={{height: 40,}}>
                <Button title = "Retry" onPress = { () => navigation.navigate("CameraScreen") }/>
            </View>
            </View>
        </View>
  );
}

const styles = StyleSheet.create({
  container: {
      flex: 1,
      flexDirection: "row",
      alignItems: "stretch",
    },
    fill: {
      width: null,
      height: null,
      flex: 1,

    },
    master: {
          flex: 1,
          backgroundColor: '#87a1b6'
        },

});
export default ImageConfirm;