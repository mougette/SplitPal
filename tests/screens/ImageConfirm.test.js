import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen, waitFor } from '@testing-library/react-native'
import {Provider as AuthProvider} from '../../src/context/AuthContext.js';
import {Context as AuthContext} from '../../src/context/AuthContext';
import {navigate} from '@reach/router'
import ImageConfirm from '../../src/screens/ImageConfirm'
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([
                                  {
                                    "TotalBalance": -85
                                  },
                                  {
                                    "FirstName": "2",
                                    "LastName": "2",
                                    "Email": "abc@e.com",
                                    "Balance": 0
                                  },
                                  {
                                    "FirstName": "fake",
                                    "LastName": "1",
                                    "Email": "fake1@gmail.com",
                                    "Balance": -19.5
                                  },
                                  {
                                    "FirstName": "fake",
                                    "LastName": "5",
                                    "Email": "fake5@gmail.com",
                                    "Balance": -23.5
                                  },
                                  {
                                    "FirstName": "hi",
                                    "LastName": "there",
                                    "Email": "newEmail@test.com",
                                    "Balance": 0
                                  },
                                  {
                                    "FirstName": "Sam",
                                    "LastName": "Hogenson",
                                    "Email": "shogenson@wisc.edu",
                                    "Balance": -40
                                  },
                                  {
                                    "FirstName": "test",
                                    "LastName": "today",
                                    "Email": "testing@test.com",
                                    "Balance": -2
                                  }
                                ]),
  })
  );
  const createTestProps = (props: Object) => ({
    navigation: {
      navigate: jest.fn()
    },
    ...props
  });
describe('Image Confirm', () => {
    const test = jest.fn();
    const Stack = createBottomTabNavigator();
    it('Select Confirm', async () => {
        const {getByText, asFragment} = render(
        <AuthProvider>
        <NavigationContainer>
        <Stack.Navigator>
        <Stack.Screen name="ImageConfirm" component={ImageConfirm} initialParams ={{stack: "group"},{itemEmail : "e@something.com"},{photo :{uri: 'https://reactnative.dev/img/tiny_logo.png'}}} />
        </Stack.Navigator>
        </NavigationContainer>
        </AuthProvider>);
        await waitFor(() => {
        fireEvent.press(getByText("Confirm"))
        expect(asFragment).toMatchSnapshot();
        })})
        it('Select Confirm', async () => {
            const {getByText, asFragment} = render(
            <AuthProvider>
            <NavigationContainer>
            <Stack.Navigator>
            <Stack.Screen name="ImageConfirm" component={ImageConfirm} initialParams ={{stack: "friend"},{itemEmail : "e@something.com"},{photo :{uri: 'https://reactnative.dev/img/tiny_logo.png'}}} />
            </Stack.Navigator>
            </NavigationContainer>
            </AuthProvider>);
            await waitFor(() => {
            fireEvent.press(getByText("Confirm"))
            expect(asFragment).toMatchSnapshot();
            })})
    it('Select Retry', async () => {
            const {getByText, asFragment} = render(
            <AuthProvider>
            <NavigationContainer>
            <Stack.Navigator>
            <Stack.Screen name="ImageConfirm" component={ImageConfirm} initialParams ={{itemEmail : "e@something.com"},{photo :{uri: 'https://reactnative.dev/img/tiny_logo.png'}}} />
            </Stack.Navigator>
            </NavigationContainer>
            </AuthProvider>);
            await waitFor(() => {
            fireEvent.press(getByText("Retry"))
            expect(asFragment).toMatchSnapshot();
            })})


});