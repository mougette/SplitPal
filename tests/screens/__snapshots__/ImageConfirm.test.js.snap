// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Confirm Select Confirm 1`] = `undefined`;

exports[`Image Confirm Select Confirm 2`] = `undefined`;

exports[`Image Confirm Select Retry 1`] = `undefined`;
